### ISSUE TITLE
Ran some basic tests

### ISSUE THREAD

**Issue by mariusvniekerk** [QuantStack/mamba#4]
`mamba create -n test2 'python>=3.6,<3.7'`

<details>
<summary>Fails with error</summary> 

```
Getting  https://conda.anaconda.org/conda-forge/linux-64
Getting  https://conda.anaconda.org/conda-forge/noarch
Getting  https://repo.anaconda.com/pkgs/main/linux-64
Getting  https://repo.anaconda.com/pkgs/main/noarch
Getting  https://repo.anaconda.com/pkgs/free/linux-64
Getting  https://repo.anaconda.com/pkgs/free/noarch
Getting  https://repo.anaconda.com/pkgs/r/linux-64
Getting  https://repo.anaconda.com/pkgs/r/noarch


Looking for: ["python[version='>=3.6,<3.7']"]


Parsing /home/mvanniekerk/miniconda3/envs/mamba/pkgs/cache/497deca9.json
60674 packages in https://conda.anaconda.org/conda-forge/linux-64
Parsing /home/mvanniekerk/miniconda3/envs/mamba/pkgs/cache/09cdf8bf.json
8923 packages in https://conda.anaconda.org/conda-forge/noarch
Parsing /home/mvanniekerk/miniconda3/envs/mamba/pkgs/cache/47929eba.json
12722 packages in https://repo.anaconda.com/pkgs/main/linux-64
Parsing /home/mvanniekerk/miniconda3/envs/mamba/pkgs/cache/3e39a7aa.json
468 packages in https://repo.anaconda.com/pkgs/main/noarch
Parsing /home/mvanniekerk/miniconda3/envs/mamba/pkgs/cache/98cb6fa3.json
13329 packages in https://repo.anaconda.com/pkgs/free/linux-64
Parsing /home/mvanniekerk/miniconda3/envs/mamba/pkgs/cache/d77296e0.json
29 packages in https://repo.anaconda.com/pkgs/free/noarch
Parsing /home/mvanniekerk/miniconda3/envs/mamba/pkgs/cache/2ce54b42.json
5057 packages in https://repo.anaconda.com/pkgs/r/linux-64
Parsing /home/mvanniekerk/miniconda3/envs/mamba/pkgs/cache/4ea078d6.json
1 packages in https://repo.anaconda.com/pkgs/r/noarch
Allowing downgrade: 1
Creating the solver...
Job: (python[version >= ='>=3.6.0.0 with python[version < ='>=3.6.0.9999 with python[version < 3.7'])

Encountered 1 problems.

Problem: package (python[version >= ='>=3.6.0.0 with python[version < ='>=3.6.0.9999 with python[version < 3.7']) does not exist
Traceback (most recent call last):
  File "/home/mvanniekerk/miniconda3/envs/mamba/bin/mamba", line 174, in <module>
    sys.exit(main())
  File "/home/mvanniekerk/miniconda3/envs/mamba/bin/mamba", line 133, in main
    to_link, to_unlink = api.solve(channel_json, installed_json_f.name, specs_seperated, strict_priority)
RuntimeError: Encountered problems while solving.
```
</details>

`mamba create -n m_test2 'python=3.6'`

<details>
<summary>Runs successfully, but installs 3.6.0 instead of 3.6.7 (latest conda-forge) or 3.6.8 (latest pkgs/main)</summary>

```
Getting  https://conda.anaconda.org/conda-forge/linux-64
Getting  https://conda.anaconda.org/conda-forge/noarch
Getting  https://repo.anaconda.com/pkgs/main/linux-64
Getting  https://repo.anaconda.com/pkgs/main/noarch
Getting  https://repo.anaconda.com/pkgs/free/linux-64
Getting  https://repo.anaconda.com/pkgs/free/noarch
Getting  https://repo.anaconda.com/pkgs/r/linux-64
Getting  https://repo.anaconda.com/pkgs/r/noarch


Looking for: ['python=3.6']


Parsing /home/mvanniekerk/miniconda3/envs/mamba/pkgs/cache/497deca9.json
60674 packages in https://conda.anaconda.org/conda-forge/linux-64
Parsing /home/mvanniekerk/miniconda3/envs/mamba/pkgs/cache/09cdf8bf.json
8923 packages in https://conda.anaconda.org/conda-forge/noarch
Parsing /home/mvanniekerk/miniconda3/envs/mamba/pkgs/cache/47929eba.json
12722 packages in https://repo.anaconda.com/pkgs/main/linux-64
Parsing /home/mvanniekerk/miniconda3/envs/mamba/pkgs/cache/3e39a7aa.json
468 packages in https://repo.anaconda.com/pkgs/main/noarch
Parsing /home/mvanniekerk/miniconda3/envs/mamba/pkgs/cache/98cb6fa3.json
13329 packages in https://repo.anaconda.com/pkgs/free/linux-64
Parsing /home/mvanniekerk/miniconda3/envs/mamba/pkgs/cache/d77296e0.json
29 packages in https://repo.anaconda.com/pkgs/free/noarch
Parsing /home/mvanniekerk/miniconda3/envs/mamba/pkgs/cache/2ce54b42.json
5057 packages in https://repo.anaconda.com/pkgs/r/linux-64
Parsing /home/mvanniekerk/miniconda3/envs/mamba/pkgs/cache/4ea078d6.json
1 packages in https://repo.anaconda.com/pkgs/r/noarch
Allowing downgrade: 1
Creating the solver...
Job: (python >= =3.6.0.0 with python < =3.6.0.9999)

Encountered 0 problems.

10 installed packages:
  - ca-certificates-2019.3.9.0-hecc5488_0.<NULL>
  - libgcc-ng-8.2.0.1-hdf63c60_1.<NULL>
  - ncurses-5.9.0.10-10.<NULL>
  - openssl-1.0.2r.0-h14c3975_0.<NULL>
  - python-3.6.0.2-2.<NULL>
  - readline-6.2.0.2-2.<NULL>
  - sqlite-3.13.0.1-1.<NULL>
  - tk-8.5.19.2-2.<NULL>
  - xz-5.2.4.1001-h14c3975_1001.<NULL>
  - zlib-1.2.11.1004-h14c3975_1004.<NULL>

Solution: 


## Package Plan ##

  environment location: /home/mvanniekerk/miniconda3/envs/mamba/envs/m_test2

  added / updated specs:
    - python=3.6


The following NEW packages will be INSTALLED:

  ca-certificates    conda-forge/linux-64::ca-certificates-2019.3.9-hecc5488_0
  libgcc-ng          pkgs/main/linux-64::libgcc-ng-8.2.0-hdf63c60_1
  ncurses            conda-forge/linux-64::ncurses-5.9-10
  openssl            conda-forge/linux-64::openssl-1.0.2r-h14c3975_0
  python             conda-forge/linux-64::python-3.6.0-2
  readline           pkgs/free/linux-64::readline-6.2-2
  sqlite             conda-forge/linux-64::sqlite-3.13.0-1
  tk                 conda-forge/linux-64::tk-8.5.19-2
  xz                 conda-forge/linux-64::xz-5.2.4-h14c3975_1001
  zlib               conda-forge/linux-64::zlib-1.2.11-h14c3975_1004


Proceed ([y]/n)? y

Preparing transaction: done
Executing transaction: done
#
# To activate this environment, use
#
#     $ conda activate m_test2
#
# To deactivate an active environment, use
#
#     $ conda deactivate
```
</details>

**Comment by mariusvniekerk** [comment]
It looks like too many zeroes get added to the version constraints.

**Comment by wolfv** [comment]
Yep, it’s a known limitation right now. You need to explicitly add a star. But also the match spec from the command line isn’t correctly fed into the mamba solver (you can see the actual condition that Libsolv gets in the console output. Should be easy to fix (or wecome!)

**Comment by wolfv** [comment]
*PR welcome :)

This line is actually from libsolv: 

Job: (python >= =3.6.0.0 with python < =3.6.0.9999)

**Comment by mariusvniekerk** [comment]
Same happens with  `mamba create -n m_test3 'python=3.6.*'`

```
Job: (python >= =3.6.0.0 with python < =3.6.0.9999)
```

**Comment by mariusvniekerk** [comment]
The other part that would be really nice is to have --dry-run print out the what it would have installed.  Makes for testing much easier

**Comment by wolfv** [comment]
It's using the same context object as conda. I was hoping dry run was
implemented on the conda stage. But it should also be trivial to check the
context object for the dry run and only print the solution!

**Comment by mariusvniekerk** [comment]
My guess is that https://github.com/wolfv/mamba/blob/master/include/parsing.cpp#L47-L59 is too aggressively applied.

**Comment by wolfv** [comment]
Well, I think fixing the spec strings given to the solve function from python would be easier. They should look exactly like the ones in the meta.yaml depend field so that they're parsed correctly. Currently, stringifying a conda match spec yields some "weird" output like python[version>=3.5] which isn't parsed correctly

**Comment by mariusvniekerk** [comment]
yeah that's probably to deal with things like `channel::package{op}{version}={string}`

**Comment by wolfv** [comment]
@mariusvniekerk it's solved thanks to using `MatchSpec(cmdline_spec).conda_build_version()`! 

We'll update the version on conda tonight.

**Comment by wolfv** [comment]
should be fixed with 0.0.2 and 0.0.3
